File: exercises/exercise5_2.js
var r = require("../out/respondent/respondent");

var es1 = r.from_interval(1000);
var es2 = r.map(r.from_interval(1000), function (event) {
  return event * 2;
});
var zipped = r.zip(es1, es2);
var token = r.subscribe(zipped, function (event) {
  console.log("Zipped values: ", event.toString());
});

setTimeout(function () {
  r.dispose(token);
}, 6000);

File: exercises/exercise5_2.clj
(ns exercise5-2
  (:require [respondent.core :as core]))

(def es1 (core/from-interval 3000))
(def es2 (core/map (core/from-interval 3000) #(* % 2)))
(def zipped (core/zip es1 es2))

(def token (core/subscribe zipped #(prn "Zipped values: " %)))

(core/dispose token)
